crates/uv/tests/pip_install.rs

```diff
@@ -184,8 +184,9 @@ fn invalid_pyproject_toml_schema() -> Result<()> {
     Ok(())
 }
 
+/// For user controlled pyproject.toml files, we enforce PEP 621.
```

Comment:
This is technically a breaking change, if we consider `pip compile`/`pip install` on a PEP 621 violating pyproject.toml a supported use case. Note that we still fall back for indirect pyproject.toml files (see test below), so e.g. a path dep on a hatch project with relative paths still works.